packages/hash/subgraph/src/stdlib/roots.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  DataTypeWithMetadata,
+  Entity,
+  EntityTypeWithMetadata,
+  GraphElement,
+  PropertyTypeWithMetadata,
+} from "../types/element";
+import { Subgraph } from "../types/subgraph";
+import { getDataTypeByEditionId } from "./element/data-type";
+import {
+  isEntityEditionId,
+  isOntologyTypeEditionId,
+} from "../types/identifier";
+import { getPropertyTypeByEditionId } from "./element/property-type";
+import { getEntityTypeByEditionId } from "./element/entity-type";
+import { getEntityByEditionId } from "./element/entity";
+import { Vertex } from "../types/vertex";
+import { mustBeDefined } from "../shared/invariant";
+
+export const getRoots = <RootType extends GraphElement>(
```

Comment:
I think this is missing JSDocs, not sure how much we care about this consistency for this method though.

Reply:
I was trying not to go overboard on investing time in docs right now, but this one could genuinely benefit from one, thanks for highlighting.
[Add JSDoc to getRoots](https://github.com/hashintel/hash/pull/1438/commits/65db69fc42a09503cda5e18b23013eac871d099d)